Main1.cpp

```diff
@@ -17,15 +18,22 @@ int main()
     cout << "Enter a Shape" << endl;
     cin >> shape;
 
-    Draw* shapeDrawer = ShapeFactory :: selectShape(shape);
+    Shape* shapeDrawer = ShapeFactory :: selectShape(shape);
```

Comment:
The function name `selectShape` is misleading. The function doesn't select the shape but in fact instantiate the object for the given input.

Also the standard name for a `Factory` pattern method is `create`.